homeassistant/helpers/template_entity.py

```diff
@@ -51,8 +51,8 @@
 
 TEMPLATE_SENSOR_BASE_SCHEMA = vol.Schema(
     {
-        vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA,
-        vol.Optional(CONF_STATE_CLASS): STATE_CLASSES_SCHEMA,
+        vol.Optional(CONF_DEVICE_CLASS): vol.Any(None, "", DEVICE_CLASSES_SCHEMA),
+        vol.Optional(CONF_STATE_CLASS): vol.Any(None, "", STATE_CLASSES_SCHEMA),
```

Comment:
We should extend the schema in `scrape` to allow `None` instead of changing it here.
Also, `""` is not a valid device or state class.